plugins/coral-plugin-love/client/LoveButton.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {Icon} from 'coral-ui';
+import styles from './styles.css';
+import {I18n} from 'coral-framework';
+import translations from './translations.json';
+import withReaction from 'coral-framework/hocs/withReaction';
+const lang = new I18n(translations);
+
+class LoveButton extends React.Component {
+  handleClick = () => {
+    const {
+      postReaction,
+      deleteReaction,
+      showSignInDialog,
+      reactionSummary
+    } = this.props;
+    const {root: {me}, comment} = this.props;
+
+    // If the current user does not exist, trigger sign in dialog.
+    if (!me) {
+      showSignInDialog();
+      return;
+    }
+
+    // If the current user is banned, do nothing.
+    if (me.status === 'BANNED') {
+      return;
+    }
+
+    if (reactionSummary) {
```

Comment:
True, but by exposing that implementation detail to plugin developers we are making it so that we can never change it without breaking plugins.

If, on the otherhand, we pass in a funciton that looks like this:

```
  const allreadyReacted = () => {
    return reactionSummary ? true : false;
  }

```

We pass that logic into the higher order component so we can change it in the future.

Also, it isn't intuitive to check for a truthy state of reactionSummary to determine whether or not a user has already taken an action. Creating a clearly named function makes it more declarative even if it does nothing but return whether or not there's a reaction summary.